### PR TITLE
Zpi 134 backend user opinions

### PR DESCRIPTION
**Pull request by JakubSamulski**
- added crud for userOpinions

### CODE REVIEW

File: backend/src/main/java/com/zpi/backend/user_opinion/UserOpinionService.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    public boolean checkIfNotRatingUsersOpinion(User user, UserOpinion userOpinion){
+        return !userOpinion.getRatingUser().equals(user);
+    }
+
+    public UserOpinion updateOpinion(Authentication authentication,long id, UpdateUserOpinionDTO updateUserOpinionDTO) throws UserDoesNotExistException, EditSomeoneElseOpinionException, OpinionDoesNotExistException, BadRequestException {
+        UpdateUserOpinionDTO.validate();
+        UserOpinion userOpinion = userOpinionRepository.findById(id).orElseThrow(() -> new OpinionDoesNotExistException("Opinion does not exist"));
+        User user = userService.getUser(authentication);
+        if(checkIfNotRatingUsersOpinion(user, userOpinion))
+            throw new EditSomeoneElseOpinionException("You can edit only your opinions");
```
Comment (pdrzazgaa) [inline backend/src/main/java/com/zpi/backend/user_opinion/UserOpinionService.java:54]: I think error messages from backend shouldn't be directed to user, but inform frontend about error, so sth like 'User can edit only his own opinion.'

File: backend/src/main/java/com/zpi/backend/user_opinion/UserOpinionService.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        userOpinion.update(updateUserOpinionDTO);
+
+        return userOpinionRepository.save(userOpinion);
+    }
+
+    public void deleteOpinion(Authentication authentication, long id) throws DeleteSomeoneElseOpinionException, UserDoesNotExistException, OpinionDoesNotExistException {
+        UserOpinion userOpinion = userOpinionRepository.findById(id).orElseThrow(()->new OpinionDoesNotExistException("Opinion does not exist"));
+        User user = userService.getUser(authentication);
+        if(checkIfNotRatingUsersOpinion(user, userOpinion))
+            throw new DeleteSomeoneElseOpinionException("You can delete only your opinions");
```
Comment (pdrzazgaa) [inline backend/src/main/java/com/zpi/backend/user_opinion/UserOpinionService.java:64]: I think error messages from backend shouldn't be directed to user, but inform frontend about error, so sth like 'User can delete only his own opinion.'

File: backend/src/main/java/com/zpi/backend/user_opinion/UserOpinionController.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    UserOpinionService userOpinionService;
+
+    @Operation(
+            summary = "Get opinions about myself",
+            description = "Returns all opinions that were given to the user"
+    )
+    @PreAuthorize("isAuthenticated()")
+    @GetMapping("/userOpinion")
+    public ResponseEntity getMineOpinions(Authentication authentication, @RequestParam int page, @RequestParam int size) throws UserDoesNotExistException {
+        System.out.println("... called getMineOpinions");
```
Comment (pdrzazgaa) [inline backend/src/main/java/com/zpi/backend/user_opinion/UserOpinionController.java:28]: getMyOpinions*

File: backend/src/main/java/com/zpi/backend/user_opinion/UserOpinionController.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+    UserOpinionService userOpinionService;
+
+    @Operation(
+            summary = "Get opinions about myself",
+            description = "Returns all opinions that were given to the user"
+    )
+    @PreAuthorize("isAuthenticated()")
+    @GetMapping("/userOpinion")
+    public ResponseEntity getMineOpinions(Authentication authentication, @RequestParam int page, @RequestParam int size) throws UserDoesNotExistException {
```
Comment (pdrzazgaa) [inline backend/src/main/java/com/zpi/backend/user_opinion/UserOpinionController.java:27]: getMyOpinions

**Review by ASerwicka — APPROVED** [review]

**Pull request merged**